### PR TITLE
Changed statre structure added all states for student

### PR DESCRIPTION
**Pull request by HurYur**
Made a lot of changes, now states and components should have right structure

### CODE REVIEW

File: client/src/app.config.router.js [hunk 1]
```diff
@@ -1,37 +1,68 @@
...
+  app.component('studentListComponent', studentListComponent);
+  app.component('teacherListComponent', teacherListComponent);
 
   app.config(['$stateProvider', '$urlRouterProvider', function($stateProvider, $urlRouterProvider) {
     $stateProvider
+
+    /* USER STUDENT STATE */
+    .state('student', {
+      url: '/student',
+      template: require('./states/student/index.html')
```
Comment (statyan) [inline client/src/app.config.router.js]: Why not a component as the other states are ?

File: client/src/components/index.js [hunk 2]
```diff
@@ -1,13 +1,8 @@
...
-    bindings: {
-      lecture: '='
-    },
-    template: require('./lectureItem.html'),
-    controller: lectureItemController,
-    controllerAs: 'lectureItem'
-  });
+
+  app.component('lectureItem', lectureItemComponent);
+  app.component('personItem', personsItemComponent);
```
Comment (statyan) [inline client/src/components/index.js]: Great idea to move components decaration out of registration to keep components/index.js clean. :+1:

File: client/src/components/lectureItem/lectureItemComponent.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const lectureItemController = require('./lectureItemController');
+
+var lectureItemComponent = {
+    bindings: {
+        lectures: '=?'
```
Comment (statyan) [inline client/src/components/lectureItem/lectureItemComponent.js]: Why this binding is an optional ? Does this component make sense without passing parameter?
Why binding is called 'lectures' (multiple), why component is used to display one lecture?

File: client/src/states/student/studentList/component.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const StudentListController = require('./controller.js');
+
+const studentListComponent = {
+    bindings: {
+        persons: '=?'
```
Comment (statyan) [inline client/src/states/student/studentList/component.js]: Should be not plural and not optional

File: client/src/states/student/studentList/controller.js [hunk 5]
```diff
@@ -2,15 +2,14 @@
 // but we need apropriate loader - scss-loader and css-loader. Check webpack loaders configuration manual
 require("./style.scss");
 
-function studentsListController($http) {
+function StudentListController($http) {
```
Comment (statyan) [inline client/src/states/student/studentList/controller.js:5]: Suggest to rewrite this onto Class notation.
```
class StudentListController {

  constructor($http) {
    this.$http = $http;
}
   $onInit() {
      this.$http.get('/api/users').then((res) => {
         this.persons = res.data;
      }
   }
}
StudentListController.$inject = ['$http']
```

**Review by statyan — CHANGES_REQUESTED** [review]

File: client/src/states/student/studentList/controller.js (hunk 5 above)
Comment (HurYur) [reply client/src/states/student/studentList/controller.js:5]: left it for future

File: client/src/states/student/studentList/component.js (hunk 4 above)
Comment (HurYur) [reply client/src/states/student/studentList/component.js]: fixed

File: client/src/components/lectureItem/lectureItemComponent.js (hunk 3 above)
Comment (HurYur) [reply client/src/components/lectureItem/lectureItemComponent.js]: got the main different, thanks
that was implemented as tested feature, and there was a lectureList instead
remade it to real lectureItem component

File: client/src/components/index.js (hunk 2 above)
Comment (HurYur) [reply client/src/components/index.js]: thanks))

File: client/src/app.config.router.js (hunk 1 above)
Comment (HurYur) [reply client/src/app.config.router.js]: made the component to it an other states

**Review by dmk1111 — APPROVED** [review]
Good job, don't forget about Class notation in future

**Pull request merged**